### ISSUE TITLE
Sqlite should raise a better exception when used as server side

### ISSUE THREAD

**Issue by TybbyAce** [Mimetis/Dotmim.Sync#275]
Hi Sebastien,

Tried to sync 2 SQLite database but got this error. Am I doing something illegal here? 

Thanks for your help.
``` cs
Dotmim.Sync.SyncException: The method or operation is not implemented. ---> System.NotImplementedException: The method or operation is not implemented.
  at Dotmim.Sync.Sqlite.SqliteScopeInfoBuilder.NeedToCreateServerScopeInfoTableAsync () [0x00000] in <33f96c3cb77c476aa3911b5a1fa9b22d>:0
  at Dotmim.Sync.CoreProvider.EnsureServerScopeAsync (Dotmim.Sync.SyncContext context, System.String scopeInfoTableName, System.Data.Common.DbConnection connection, System.Data.Common.DbTransaction transaction, System.Threading.CancellationToken cancellationToken, System.IProgress`1[T] progress) [0x0003d] in <d2f8f5ce55ab4a8892c9b514592a8f17>:0
  at Dotmim.Sync.RemoteOrchestrator.EnsureSchemaAsync (System.Threading.CancellationToken cancellationToken, System.IProgress`1[T] progress) [0x0034b] in <d2f8f5ce55ab4a8892c9b514592a8f17>:0
  --- End of inner exception stack trace ---
  at Dotmim.Sync.BaseOrchestrator.RaiseError (System.Exception exception) [0x00046] in <d2f8f5ce55ab4a8892c9b514592a8f17>:0
  at Dotmim.Sync.RemoteOrchestrator.EnsureSchemaAsync (System.Threading.CancellationToken cancellationToken, System.IProgress`1[T] progress) [0x00f0a] in <d2f8f5ce55ab4a8892c9b514592a8f17>:0
  at Dotmim.Sync.SyncAgent.SynchronizeAsync (Dotmim.Sync.Enumerations.SyncType syncType, System.Threading.CancellationToken cancellationToken, System.IProgress`1[T] progress) [0x00d28] in <d2f8f5ce55ab4a8892c9b514592a8f17>:0
  at AnotherSync.MainPage.SyncDB () [0x0007f] in /Users/Tibor/Projects/AnotherSync/AnotherSync/MainPage.xaml.cs:89
```

``` cs

 public static async Task SyncDB()
        {
            SQLitePCL.Batteries_V2.Init();

            SqliteSyncProvider ServerProvider = new SqliteSyncProvider(Db1PathAndName);
            SqliteSyncProvider ClientProvider = new SqliteSyncProvider(Db2PathAndName);

            // Tables involved in the sync process:
            var tables = new string[] { "T1" };

            // Sync agent
            SyncAgent agent = new SyncAgent(ServerProvider, ClientProvider, tables);
            var result = await agent.SynchronizeAsync();
        }


```

**Comment by Mimetis** [comment]
`SqliteSyncProvider`  can not be used as a **Server** provider.

Anyway, you should have a better exception raised, to have a better comprehension.
Will look into it

**Comment by Mimetis** [comment]
by the way, I have edited your message to format your code.
Please use the markdown formatting to have a better reading message :)

**Comment by Mimetis** [comment]
I have edited your title as well to reflect the good issue

**Comment by TybbyAce** [comment]
Thanks, Sebastien. Sorry for the code, I am such a rookie (literally) :)

**Comment by Mimetis** [comment]
resolved in `v0.5.5-beta-0411`
You can test it with the prerelease nuget packages https://www.nuget.org/packages/Dotmim.Sync.Core/0.5.5-beta-0411

Let me know if it's actually resolved for you.
if not, I will re-open the issue.
